internal/sftp/goclient.go

```diff
@@ -64,23 +64,29 @@ func (c *GoClient) Upload(ctx context.Context, src io.Reader, dest string) (int6
 	return bytes, remotePath, nil
 }
 
-// Delete removes the data from dest, a new SFTP connection is opened before
+// Delete removes the data from dest. A new SFTP connection is opened before
 // removing the file, and closed when the delete is complete.
 func (c *GoClient) Delete(ctx context.Context, dest string) error {
 	if err := c.dial(); err != nil {
 		return fmt.Errorf("SFTP: unable to dial: %w", err)
 	}
 	defer c.close()
 
-	// Delete the file
-	if err := c.sftp.Remove(dest); err != nil {
-		if os.IsNotExist(err) {
-			return fmt.Errorf("SFTP: file does not exist: %w", err)
-		} else if os.IsPermission(err) {
-			return fmt.Errorf("SFTP: insufficient permissions to delete file: %w", err)
+	// SFTP assumes that "/" is used as the directory separator. See:
+	// https://datatracker.ietf.org/doc/html/draft-ietf-secsh-filexfer-02#section-6.2
+	remotePath := strings.TrimSuffix(c.cfg.RemoteDir, "/") + "/" + dest
+
+	if err := c.sftp.Remove(remotePath); err != nil {
+		head := fmt.Sprintf("SFTP: unable to remove file %q", dest)
+		if os.IsNotExist(err) || os.IsPermission(err) {
+			return fmt.Errorf("%s: %w", head, err)
+		}
+		if status, ok := err.(*sftp.StatusError); ok {
+			return fmt.Errorf("%s: %v (status code: %d)", head, status.FxCode(), status.Code)
 		}
-		return fmt.Errorf("SFTP: unable to remove file %q: %v", dest, err)
+		return fmt.Errorf("%s: %v", dest, err)
```

Comment:
This shouldn't happen:

    SFTP: unable to remove file "test.txt": ...

Reply:
It *shouldn't* happen, but if it did happen I think `return fmt.Errorf("SFTP: remove: %s: %v", dest, err)` (or something similar) would provide helpful context for tracking down where the error came from in the code.

Reply:
Yes, good point, fixed in d89087b2.